frontend/server/controllers/CourseController.php

```diff
@@ -298,25 +309,52 @@ public static function apiCreate(Request $r) {
         self::authenticateRequest($r);
         self::validateCreate($r);
 
-        if ($r['alias'] == 'new') {
+        self::createCourseAndGroup(new Courses([
+            'name' => $r['name'],
+            'description' => $r['description'],
+            'alias' => $r['alias'],
+            'school_id' => $r['school_id'] ?? null,
+            'start_time' => gmdate('Y-m-d H:i:s', $r['start_time']),
+            'finish_time' => gmdate('Y-m-d H:i:s', $r['finish_time']),
+            'public' => is_null($r['public']) ? false : $r['public'],
+            'show_scoreboard' => boolval($r['show_scoreboard']),
+            'needs_basic_information' => $r->ensureBool('needs_basic_information', false /*isRequired*/),
+            'requests_user_information' => $r['requests_user_information'],
+        ]), $r['current_user_id']);
+
+        return ['status' => 'ok'];
+    }
+
+    /**
+     * Function to create a new course with its corresponding group
+     *
+     * @param Courses $course
+     * @param $creatorUserId
+     * @return Courses
+     */
+    private static function createCourseAndGroup(
+        Courses $course,
+        int $creatorUserId
+    ) : Courses {
+        if ($course->alias == 'new') {
             throw new DuplicatedEntryInDatabaseException('aliasInUse');
         }
 
-        if (!is_null(CoursesDAO::getByAlias($r['alias']))) {
+        if (!is_null(CoursesDAO::getByAlias($course->alias))) {
             throw new DuplicatedEntryInDatabaseException('aliasInUse');
         }
 
         DAO::transBegin();
 
         $group = GroupController::createGroup(
-            $r['alias'],
-            'students-' . $r['alias'],
-            'students-' . $r['alias'],
-            $r['current_user_id']
+            $course->alias,
+            'students-' . $course->alias,
```

Comment:
nit:

```suggestion
            "students-{$course->alias}",
```

igual abajo.

Reply:
Listo